recipes/libnvshmem/meta.yaml

```
@@ -0,0 +1,225 @@
+        - cudatoolkit       # [(cuda_compiler_version or "").startswith("11")]
+        - ucx
+      run:
+        - {{ pin_compatible("cuda-version", min_pin="x", max_pin="x") }}
+        - cuda-cudart       # [(cuda_compiler_version or "").startswith("12")]
+        - libpciaccess
+        - libpmix
+        - nccl
+        - pmix-bin
+        - ucx
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -L $PREFIX/lib/libnvshmem_host.so.{{ soname }}
+        - test -f $PREFIX/lib/libnvshmem_host.so.{{ version }}
+        - test -L $PREFIX/lib/nvshmem_bootstrap_mpi.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_bootstrap_mpi.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_bootstrap_pmi.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_bootstrap_pmi.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_bootstrap_pmi2.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_bootstrap_pmi2.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_bootstrap_pmix.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_bootstrap_pmix.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_bootstrap_shmem.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_bootstrap_shmem.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_bootstrap_uid.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_bootstrap_uid.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_transport_ibdevx.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_transport_ibdevx.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_transport_ibgda.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_transport_ibgda.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_transport_ibrc.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_transport_ibrc.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_transport_libfabric.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_transport_libfabric.so.{{ soname }}.0.0
+        - test -L $PREFIX/lib/nvshmem_transport_ucx.so.{{ soname }}
+        - test -f $PREFIX/lib/nvshmem_transport_ucx.so.{{ soname }}.0.0
+
+  - name: libnvshmem-dev
+    build:
+      ignore_run_exports_from:    # [(cuda_compiler_version or "").startswith("12")]
+        - {{ compiler('cuda') }}  # [(cuda_compiler_version or "").startswith("12")]
+        - cuda-cudart-dev         # [(cuda_compiler_version or "").startswith("12")]
+      missing_dso_whitelist:
+        - "*libcuda.so*" # driver
+        - "*libnvshmem_host.so.{{ soname }}" # avoids: ERROR (libnvshmem-dev,bin/perftest/device/pt-to-pt/shmem_put_bw): lib/libnvshmem_host.so.3 not found in any packages. We DO test for its existence though.
+        - "*libmpi*"
+        - "*libpmix*"
+        - "*liboshmem*"
+      run_exports:
+        - {{ pin_subpackage("libnvshmem" ~ soname, max_pin=None) }}
```

```suggestion
      run_exports:
        - {{ pin_subpackage("libnvshmem", max_pin=None) }}
        - {{ pin_subpackage("libnvshmem" ~ soname, max_pin=None) }}
```
If you want a non-ABI-versioned metapackage. part 2 of 2.

---

I won't bother with this after your explanation of how OS searches for ABI libs.